1-exercises/D-array-filter/exercise.js

```diff
@@ -8,7 +8,11 @@
 
 let pairsByIndexRaw = [[0, 3], [1, 2], [2, 1], null, [1], false, "whoops"];
 
-let pairsByIndex; // Complete this statement
+let pairsByIndex = pairsByIndexRaw.filter((e) => {
+  if (typeof e === "object" && e !== null) {
+    return e.length === 2;
+  }
+}); // Complete this statement
```

Comment:
Although it is hard to name things properly, please try to avoid single character variable names.

My version looked like this:
```javascript
let pairsByIndex = pairsByIndexRaw.filter((item) => {
  if (item === undefined) return false;
  if (item === null) return false;
  if (typeof(item) !== "object") return false;
  if (item.length !== 2) return false;
  return true;
});
```
My reason for writing the function like this was to be explicit about the validation rules in the function. Your version is also fine.